test/test_tags.jl

```diff
@@ -0,0 +1,125 @@
+"Tags starting with this identifier are broken. Remove upon fix."
+const brokenidentifiers = (
+        "deprecated_returnsquarepower",
+        "deprecated_squarepower",
+        "DocTestSetup",
+        "printgreeting",
+        "returnsquarepower",
+)
+"Kinds that have not been added yet (and are therefore broken). Remove upon fix."
+const brokenkinds = ('m', 's')
+
+"Path to Ctags configuration file (the parser)."
+const ctagsconfigpath = joinpath(@__DIR__, "..", "ctags")
+"Test file to generate the tags from."
+const testfilepath = joinpath(@__DIR__, "testfile.jl")
+"File containing test tags to compare against."
+const testtagspath = joinpath(@__DIR__, "testtags")
+
+"Ctags binaries to sequentially try to call if one is not found."
+const ctagsbins = (
+        "uctags",
+        "ectags",
+        "universal-ctags",
+        "exuberant-ctags",
+        "u-ctags",
+        "e-ctags",
+        "ctags",
+)
+"Arguments to pass to a Ctags binary."
+const ctagsargs = (
+        "--options=$ctagsconfigpath",  # Include our tags file.
+        "-f -"  # Write to stdout.
+)
+
+
+"Utility function to convert an `AbstractVector` of tags to a common format."
+function converttags(tags::AbstractVector)
+    tags = filter(s -> !isempty(s) && s[1] != '!', tags)
+    Set(tags)
+end
+
+"Return tags generatad from the given file and convert them using [`converttags`](@ref)."
+function generatetags(testfile::AbstractString, ctagsbin::AbstractString="ctags")
+    ctagscmd = `$ctagsbin $ctagsargs $testfile`
+    tags = open(ctagscmd, "r", stdout) do io
+        readlines(io)
+    end
+    return converttags(tags)
+end
+
+function trygeneratetags(testfile::AbstractString)
+    for ctagsbinary in ctagsbins
+        try
+            return generatetags(testfile, ctagsbinary)
+        catch e
+            # For compatibility accept either exception type.
+            if !(err isa LoadError || err isa ErrorException
```

Comment:
For uctags the `generatetags` method failed with a `Base.IOError` instead of a `LoadError` (although from the printed error message it looked like a `LoadError`), so you should add that one to the list of 'acceptable' error.

I'm not sure if adding `LoadError` to the list of acceptable Exceptions is necessary. At least for my setup none of the failing tag binaries did throw an actual `LoadError`.

Reply:
I expected `ProcessFailedException`

Reply:
Sorry, I was busy when pushing and didn't test. I'll be sure to fix it, thanks again, Karl!
I actually can't remember why I included the `LoadError` (should have documented that one) but I'm sure it's there for a reason. Maybe I can replicate it again.

Hey Lyndon, that one's listed one line below, you can't see it in the diff.

Reply:
ah cool cool